tests/federation/test_federation_client.py

```diff
@@ -230,21 +227,3 @@ def _get_pdu_once(self) -> EventBase:
         self.assertEqual(remote_pdu.internal_metadata.outlier, False)
 
         return remote_pdu
-
-
-def _mock_response(resp: JsonDict):
```

Comment:
Because we already had a `FakeResponse` class and I enhanced it, I took the opportunity to replace this here